### PR TITLE
apis: add compaction config for TiKV

### PR DESCRIPTION
**Pull request by CheneyDing**
### What problem does this PR solve?

Add some compaction config for TiKV.
### What is changed and how does it work?

add item in TiKV.
### Code changes

- [x] Has Go code change
- [x] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [x] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
None
```

### CODE REVIEW

**Review by nolouch — COMMENTED** [review]
cc @lonng @DanielZhangQD

File: pkg/apis/pingcap/v1alpha1/tikv_config.go [hunk 1]
```diff
@@ -504,6 +504,8 @@ type TiKVCfConfig struct {
 	// +optional
 	EnableDoublySkiplist *bool `json:"enable-doubly-skiplist,omitempty" toml:"enable-doubly-skiplist,omitempty"`
 	// +optional
+	EnableCompactionGuard *bool `json:"enable-compaction-guard,omitempty" toml:"enable-compaction-guard,omitempty"`
```
Comment (nolouch) [inline pkg/apis/pingcap/v1alpha1/tikv_config.go:507]: ref https://github.com/tikv/tikv/pull/8115, plz add
```
compaction-guard-min-output-file-size
compaction-guard-max-output-file-size
```

**Comment by DanielZhangQD** [comment]
@CheneyDing @nolouch We have supported TOML format config from TiDB Operator v1.1.6, please check detail in https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/configure-a-tidb-cluster#%E9%85%8D%E7%BD%AE-tikv-%E9%85%8D%E7%BD%AE%E5%8F%82%E6%95%B0, there is no need to update the config struct for different components anymore.

**Comment by CheneyDing** [comment]
OK, got it, thanks!

**Pull request closed without merging**